hpy/debug/src/debug_ctx.c

```diff
@@ -0,0 +1,121 @@
+#include <string.h>
+#include <stdio.h>
+#include "debug_internal.h"
+#include "autogen_debug_ctx_init.h"
+
+static struct _HPyContext_s g_debug_ctx = {
+    .name = "HPy Debug Mode ABI",
+    ._private = NULL,
+    .ctx_version = 1,
+};
+
+// NOTE: at the moment this function assumes that uctx is always the
+// same. If/when we migrate to a system in which we can have multiple
+// independent contexts, this function should ensure to create a different
+// debug wrapper for each of them.
+static int debug_ctx_init(HPyContext dctx, HPyContext uctx)
+{
+    if (dctx->_private != NULL) {
+        // already initialized
+        assert(get_info(dctx)->uctx == uctx); // sanity check
+        return 0;
+    }
+    // initialize debug_info
+    // XXX: currently we never free this malloc
+    HPyDebugInfo *info = malloc(sizeof(HPyDebugInfo));
+    if (info == NULL) {
+        HPyErr_NoMemory(uctx);
+        return -1;
+    }
+    info->magic_number = HPY_DEBUG_MAGIC;
+    info->uctx = uctx;
+    info->current_generation = 0;
+    info->open_handles = NULL;
+    //info->closed_handles = NULL;
```

Comment:
Is there a plan to used `closed_handles` later?

Reply:
yes. It will be used to detect when you try to use a handle which has already been closed